src/api/controllers/articleRatingsController.js

```diff
@@ -0,0 +1,99 @@
+import _ from 'lodash';
+import models, { sequelize } from '../models/index';
+import status from '../../helpers/constants/status.codes';
+import sendError from '../../helpers/error.sender';
+import calculateRatingsPercentage from '../../helpers/calculate.ratings.percentages';
+import errorMessages from '../../helpers/constants/error.messages';
+import generatePaginationDetails from '../../helpers/generate.pagination.details';
+
+const { Rating, User } = models;
+
+/**
+ * containing all controllers of the ratings process
+ *
+ * @export
+ * @class ratings
+ */
+export default class articleRatings {
+  /**
+   * A controller to calculate the ratings percentages,
+   *
+   * @param {Object} req - the request object
+   * @param {Object} res - the result object
+   * @returns {Object} res
+   */
+  static async getArticleRatingStatistics(req, res) {
+    // Initialising variables
+    const result = {};
+    const { slug } = req.params;
+
+    // Fetching the articles count per rating
+    const ratingsCount = await Rating.findAll({
+      where: {
+        articleSlug: slug
+      },
+      group: ['rating'],
+      attributes: ['rating', [sequelize.fn('COUNT', 'TagName'), 'count']]
+    });
+
+    // Sending the result
+    if (_.isEmpty(ratingsCount)) {
+      // Error if there are no ratings for the article
+      sendError(status.NOT_FOUND, res, 'ratings', errorMessages.ratingsNotFound);
+      result.status = status.NOT_FOUND;
+    } else {
+      // Calculating the percentages
+      const ratingsWithPercentages = calculateRatingsPercentage(ratingsCount);
+
+      // sending the result back
+      result.status = status.OK;
+      result.data = ratingsWithPercentages;
+      res.status(status.OK).json(result);
+    }
+  }
+
+  /**
+   * A controller to get users for a given rating
+   * with pagination support
+   *
+   * @param {Object} req - the request object
+   * @param {Object} res - the result object
+   * @returns {Object} res
+   */
+  static async getRatingUsers(req, res) {
+    // Initialising variables
+    const result = {};
+    const { offset, limit } = req.query;
+    const { slug, rating } = req.params;
+
+    // Fetching data from the database
+    const ratingUsers = await User.findAndCountAll({
+      attributes: ['id', 'username', 'email', 'firstName', 'lastName'],
+      include: [
+        {
+          model: Rating,
+          where: {
+            articleSlug: slug,
+            rating
+          },
+          attributes: []
+        }
+      ],
+      offset,
```

Comment:
@nkpremices let's give offset and limit default values e.g `offset || 0, limit || 10`

Reply:
noted @MemunaHaruna

Reply:
I added them in the validation schema